src/docs-developer-guide.rst

```diff
@@ -122,3 +122,221 @@ For example:
+  :param str err: Name of the test
+  :param str msg: Error or warning message
+  :param bool ignore_case: For using :py:data:`re.IGNORECASE`
+  :param float max_errors: Maximum number of errors to be generated
+  :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+  :rtype: list
+
+
+.. py:function:: consistency_check(text, word_pairs, err, msg, offset=0)
+
+   To check for consistency for the given word pairs.
+   The parameters :py:data:`offset`
+   is not needed for style guide testing.
+
+   :param str text: Text to be checked
+   :param list word_pairs: Word pairs to be checked for consistency
+   :param str err: Name of the test
+   :param str msg: Error or warning message
+   :return: The error list consisting of error tuples: :py:const:`[(start, end, err, msg, replacement)]`.
+   :rtype: list
+
+.. note::
+
+  The checker functions are used by the
+  built-in proselint function :py:func:`lint`
+  to generate an error list of different format.
+  The returned list finally is: :py:const:`[(check, message, line, column, start, end, end - start, "warning", replacements)]`
+
+.. seealso:: `Proselint source code <https://github.com/amperser/proselint/blob/master/proselint/tools.py>`_
+
+.. rubric:: Example Usage
+
+.. code-block:: python
+
+  @memoize
+  def example(text):
+      """Example check."""
+      err = "style-guide.example"
+      msg = "A demonstration for writing checks."
+      regex = "[\.\?!](example)"
+
+      return existence_check(text, [regex], err, msg, ignore_case=False, 
+                         require_padding=False)
+
+When you define code-blocks which
+use built-in proselint testing,
+specify the class **style-checks**.
+
+.. code-block:: rst
+
+  .. code-block:: python
+    :class: style-checks
+
+The generated file after parsing code
+for style checks is :file:`style-checks.py`.
+
+
+If the test is too large
+to be defined in the file :file:`docs-style-guide.rst`,
+you can use a snippet from the test
+(as :ref:`here <american-spelling>`).
+The code-blocks for such snippets
+should specify the class **proselint-extra-checks**.
+Define the complete test
+in the file :file:`/style-guide/proselint-extra-checks.py`.
+
+.. _independent-checks:
+
+Independent checks
+~~~~~~~~~~~~~~~~~~~
+
+Apart from the checks, which are to be run through proselint,
+you can add extra checks to be run independently.
+They are not enabled in :file:`proselintrc` as well.
+For example, the checks for finding quote marks and section labels
+do not use any built-in functions to obtain an error list.
+
+.. rubric:: Example Usage
+
+.. code-block:: python
+
+  def check_quotes(text):
+      """Avoid using straight quotes."""
+      err = "style-guide.check-quote"
+      msg = "Avoid using quote marks."
+      regex = r"\"[a-zA-z0-9 ]{1,15}\""
+
+      errors = []
+
+      for m in re.finditer(regex, text):
```

Comment:
What does m represent here?

Reply:
Oh --- I bet it's match object, right?

Reply:
Oh yes, my bad with the naming. I will fix that!

Reply:
Not a huge deal, but its helpful to be explicit. 
This is always true, and is doubly true in Open Source,
where you never know who the next person to touch your code will be.

Reply:
That is so right. A cleaner code always makes contributions easier!

Reply:
I have fixed the naming!